cardano-cli/test/Test/CLI/Shelley/Golden/Address/Build.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Test.CLI.Shelley.Golden.Address.Build
+  ( golden_shelleyAddressBuild
+  ) where
+
+import Cardano.Prelude hiding (to)
+
+import Hedgehog (Property, (===))
+
+import qualified Data.List as L
+import qualified Test.OptParse as OP
+
+{- HLINT ignore "Use camelCase" -}
+
+golden_shelleyAddressBuild :: Property
+golden_shelleyAddressBuild = OP.propertyOnce $ OP.workspace "tmp/address-key-gen" $ \tempDir -> do
+  addressVKeyFile <- OP.noteEvalM $ OP.newFileWithContents (tempDir <> "/address.vkey") "{\
+    \    \"type\": \"PaymentVerificationKeyShelley\",\
```

Comment:
I think this would be more readable if we used a golden file instead (`cardano-cli/test/Test/golden/shelley/keys/payment_keys/verification_key`). This is a personal preference however, I'd like to hear what @erikd & @intricate thinks as well.

Reply:
I did it this way to avoid having to find a home for the files, but if you can give me some guidance on how to place files relative the module or project, I can switch it up.

Reply:
Yeah, this stuff should defnitely be in a files checked into git.

Pretty sure some of the other tests already do this.

Reply:
@newhoggy so in this case, you can use read the contents of  `cardano-cli/test/Test/golden/shelley/keys/payment_keys/verification_key` and feed it to `newFileWithContents`. There should be existing golden files for the tests you are writing in this PR. If not, create the relevant golden file and put them in `golden/`. We can then place them appropriately within `golden/` if need be afterwards

Reply:
I agree with using a golden file.

Reply:
I placed the golden files here:

* `cardano-cli/test/Test/golden/shelley/addresses/staking-address.hex`
* `cardano-cli/test/Test/golden/shelley/addresses/enterprise-address.hex`